lua/rocks/internal-types.lua

```diff
@@ -0,0 +1,22 @@
+---@mod rocks.types
+--
+-- Copyright (C) 2023 NTBBloodbath
+--
+-- Version:    0.1.0
+-- License:    GPLv3
+-- Created:    05 Jul 2023
+-- Updated:    27 Aug 2023
+-- Homepage:   https://github.com/nvim-neorocks/rocks.nvim
+-- Maintainers: NTBBloodbath <bloodbathalchemist@protonmail.com>, Vhyrro <vhyrro@gmail.com>
+--
+---@brief [[
+--
+-- rocks.nvim internal type definitions.
+--
+---@brief ]]
+
+---@class (exact) Rock
+---@field public name string
+---@field public version string
```

Comment:
This is a WIP feature of lua-language-server that will allow more strict type definitions (e.g., creating records with undefined fields would be invalid).

It doesn't work properly yet, but it will likely turn out to be useful in the future.